cfn-resources/federated-database-instance/mongodb-atlas-federateddatabaseinstance.json

```diff
@@ -0,0 +1,355 @@
+        },
+        "list": {
+            "permissions": ["secretsmanager:GetSecretValue"]
+        }
+    },
+    "properties": {
+        "CloudProviderConfig": {
+            "type": "object",
+            "description": "Cloud provider linked to this data lake.",
+            "$ref": "#/definitions/CloudProviderConfig"
+        },
+        "DataProcessRegion": {
+            "type": "object",
+            "description": "Information about the cloud provider region to which the data lake routes client connections. MongoDB Cloud supports AWS only.",
+            "$ref": "#/definitions/DataProcessRegion"
+        },
+        "ProjectId": {
+            "type": "string",
+            "description": "Unique 24-hexadecimal digit string that identifies your project.",
+            "maxLength": 24,
+            "minLength": 24,
+            "pattern": "^([a-f0-9]{24})$"
+        },
+        "TenantName": {
+            "type": "string",
+            "description": "Human-readable label that identifies the data federation."
+        },
+        "SkipRoleValidation": {
+            "type": "boolean",
+            "description": "Flag that indicates whether this request should check if the requesting IAM role can read from the S3 bucket. AWS checks if the role can list the objects in the bucket before writing to it. Some IAM roles only need write permissions. This flag allows you to skip that check."
+        },
+        "Storage": {
+            "type": "object",
+            "description": "Configuration information for each data store and its mapping to MongoDB Cloud databases.",
+            "$ref": "#/definitions/Storage"
+        },
+        "State": {
+            "type": "string",
+            "description": "Type of Federated Database Instances to return."
+        },
+        "HostNames": {
+            "type": "array",
+            "insertionOrder": false,
+            "description": "Type of Federated Database Instances to return.",
+            "items": {
+                "type": "string"
+            }
+        },
+        "Profile": {
+            "type": "string",
+            "description": "The profile is defined in AWS Secret manager. See [Secret Manager Profile setup](../../../examples/profile-secret.yaml).",
+            "default": "default"
+        }
+    },
+    "readOnlyProperties": [
+        "/properties/CloudProviderConfig/ExternalId",
+        "/properties/CloudProviderConfig/IamAssumedRoleARN",
+        "/properties/CloudProviderConfig/IamUserARN",
+        "/properties/HostNames",
+        "/properties/State",
+        "/properties/TenantName"
+    ],
+    "writeOnlyProperties": [
+        "/properties/CloudProviderConfig/TestS3Bucket"
+    ],
+    "createOnlyProperties": [
+        "/properties/ProjectId",
+        "/properties/Profile"
+    ],
+    "required": [
+        "ProjectId"
+    ],
+    "primaryIdentifier": [
+        "/properties/ProjectId",
+        "/properties/TenantName"
```

Comment:
did you tested the read with a CFN stack?? i see that the profile is missing on the primaryIdentifier

Reply:
Yes I tested with the example that is added, if we keep the Profile in PrimaryIdentifier we have to add return this as part of Get API which may not be required all the times to the user.

Reply:
i think you are confusing the primaryIdentifier with the readOnly, primaryIdentifiers can be createOnly and readOnly, if you set them to readOnly the user will be able to output them.
Also remember that only the primary identifiers reaches the read, that is why we added the profile, on your case is working because it arrives on nil and then is set as the default here..
<img width="662" alt="image" src="https://github.com/mongodb/mongodbatlas-cloudformation-resources/assets/69170650/882eab33-dca2-42b4-8aad-7a37159192a0">

if the user provides another profile, it will always use the default, remember that this is a behaviour that we had to fix on the end on the phase 1

Reply:
I moved Profile to CreateOnly Properties and tested.

<img width="779" alt="image" src="https://github.com/mongodb/mongodbatlas-cloudformation-resources/assets/15726683/1677b0b7-d89c-4b90-9d06-ea07a9b57cf2">